hid-xpadneo/src/hid-xpadneo.c

```diff
@@ -850,7 +888,7 @@ static int xpadneo_raw_event(struct hid_device *hdev, struct hid_report *report,
 	}
 
 	/* correct button mapping of Xbox controllers in Linux mode */
-	if (report->id == 1 && (reportsize == 17 || reportsize == 39)) {
+	if (report->id == 1 && (reportsize == 17 || reportsize == 39 || reportsize == 55)) {
```

Comment:
You probably need to make a variant of this in a second if block for the paddle buttons. Ordering may be important if you check the same bytes, as we replace bytes in `data[]` in-place.